specifyweb/frontend/js_src/lib/hooks/useResourceValue.tsx

```diff
@@ -132,9 +132,7 @@ export function useResourceValue<
       );
       if (field === undefined) return;
 
-      // This assumes that there are no field blockers set by anything else
-      if (parseResults.isValid) setBlockers([]);
-      else setBlockers([parseResults.reason]);
+      if (!parseResults.isValid) setBlockers([parseResults.reason]);
```

Comment:
would this change prevent blockes from being removed from a field once user resolved the error?

perhaps a better solution would to modify `setBlockers([]);` to instead be something like this?

```ts
setBlockers(blockers=>blockers.filter(blocker=>previousBlockersRef.current.includes(blocker)));
```

that would act just like fieldBlockerKey in many cases, without the associated complexities

Reply:
Using a `previousBlockersRef` directly as a React ref would be the optimal solution, but is not currently possible as the same ref would need to be used in a non-React component: https://github.com/specify/specify7/blob/29e6646431665b1fe432d36e4cec181bcea0ae08/specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts#L165

In the meantime I have implemented a solution proposed by @realVinayak in https://github.com/specify/specify7/pull/4588/commits/076fc3cf2b043158ea682bbd0797f9f9210db0c4. 

We could still likely implement the 'ref' idea in theory, although it would need to handle the case of running in non-React functions.